banyand/liaison/grpc/registry.go

```diff
@@ -435,7 +437,37 @@ type topNAggregationRegistryServer struct {
 func (ts *topNAggregationRegistryServer) Create(ctx context.Context,
 	req *databasev1.TopNAggregationRegistryServiceCreateRequest,
 ) (*databasev1.TopNAggregationRegistryServiceCreateResponse, error) {
-	if err := ts.schemaRegistry.TopNAggregationRegistry().CreateTopNAggregation(ctx, req.GetTopNAggregation()); err != nil {
+	topNSchema := req.GetTopNAggregation()
```

Comment:
Move the measure creation operation into `topNProcessorManager.start`. Liaison is a proxy and should not implement complicated functions.

If there is no such measure, the manager should create a new one. The manager should check the on-disk measure's specs. If it's different from expectation, it could get updated.

Reply:
This part is moved to `measure` pkg. PTAL

Reply:
We cannot do this since `topNProcessorManager` will start in the `PreRun` stage, but the measure notifier is registered in the `Serve` stage which will not be called if new measures are create.....Any idea?

Reply:
I think we should avoid creating new measures while loading.......

Reply:
Placing the creation in the loading phase ensures the measure always exists. Restarting the server will create the target measure if it's missing. As I mentioned, banyandb doesn't support transactional operations. Creating more than one resource in a task is too dangerous.

There are two options here:

1. Move `processorManager.start` to the `Serve` phase. That means you should iterate all measures to start their embedded `processorManager`
2. Introduce Etcd's transactional operation. Move the creation operation to the `metadata` module instead of `liaison`.

Reply:
> 1. Move `processorManager.start` to the `Serve` phase. That means you should iterate all measures to start their embedded `processorManager`

I take the first approach which is straight forward.

> 2. Introduce Etcd's transactional operation. You could leverage a transaction to create topN resource and its derived measure by moving the creation operation to the `metadata` module instead of `liaison`.

We should support transactional op which would be helpful for https://github.com/apache/skywalking/issues/10010. The current client impl could cause incomplete metadata registration in some edge cases.

Reply:
We might support the transactional operations on fixing 10010